module/data/region-behavior/difficult-terrain.mjs

```diff
@@ -0,0 +1,87 @@
+const { BooleanField, NumberField, SetField, StringField } = foundry.data.fields;
+
+/**
+ * The data model for a region behavior that represents an area of difficult terrain.
+ */
+export default class DifficultTerrainRegionBehaviorType extends foundry.data.regionBehaviors.RegionBehaviorType {
+
+  /** @override */
+  static LOCALIZATION_PREFIXES = ["DND5E.REGIONBEHAVIORS.DIFFICULTTERRAIN"];
+
+  /* ---------------------------------------- */
+
+  /** @override */
+  static defineSchema() {
+    const dispositions = { ...foundry.applications.sheets.TokenConfig.TOKEN_DISPOSITIONS };
+    delete dispositions[CONST.TOKEN_DISPOSITIONS.SECRET];
+    return {
+      magical: new BooleanField(),
+      types: new SetField(new StringField({ choices: () => CONFIG.DND5E.difficultTerrainTypes })),
```

Comment:
We typically can't use `choices` where the choices come from `CONFIG`. It would mean that the region behaviour gets marked as invalid if a key is deleted the config, or a module that was adding a key is disabled. Maybe that's ok for region behaviours, but it's not something we do elsewhere in the system.

Reply:
I'll stick with the custom sheet for consistency with how we handled choices across the system. This would be a good time to support `choices` one data fields that aren't validated, but are available for generated inputs (and possibly AE suggestions in the future).